bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/TriggerLocationCompactService.java

```diff
@@ -0,0 +1,154 @@
+            String output = "Not trigger Entry Location RocksDB compact.";
+
+            if (StringUtils.isBlank(requestBody)) {
+                output = "Empty request body";
+                response.setBody(output);
+                response.setCode(HttpServer.StatusCode.BAD_REQUEST);
+                return response;
+            }
+
+            try {
+                @SuppressWarnings("unchecked")
+                Map<String, Object> configMap = JsonUtil.fromJson(requestBody, HashMap.class);
+                Boolean isEntryLocationCompact = (Boolean) configMap
+                        .getOrDefault("entryLocationRocksDBCompact", false);
+                String entryLocations = (String) configMap.getOrDefault("entryLocations", "");
+                List<String> entryLocationDBPath = ledgerStorage.getEntryLocationDBPath();
+                if (!isEntryLocationCompact) {
+                    // If entryLocationRocksDBCompact is false, doing nothing.
+                    response.setBody(output);
+                    response.setCode(HttpServer.StatusCode.OK);
+                    return response;
+                }
+                if (StringUtils.isNotBlank(entryLocations)) {
+                    // Specified trigger RocksDB compact entryLocations.
+                    List<String> locations = Arrays.asList(entryLocations.trim().split(","));
+                    Map<String, Boolean> compactStatus = ledgerStorage.isEntryLocationCompacting(locations);
+
+                    if (entryLocationDBPath.containsAll(locations) && !compactStatus.containsValue(true)) {
+                        ledgerStorage.entryLocationCompact(locations);
+                        output = String.format("Triggered entry Location RocksDB: %s compact on bookie:%s.",
+                                entryLocations, bookieServer.getBookieId());
+                        response.setCode(HttpServer.StatusCode.OK);
+                    } else if (entryLocationDBPath.containsAll(locations)) {
+                        // If any specified entry location RocksDB is running, not trigger compact again.
+                        output = String.format("Specified entry location RocksDB: %s already running compact,"
+                                + " compact status: %s.", entryLocations, compactStatus);
+                        response.setCode(HttpServer.StatusCode.OK);
+                    } else {
+                        output = String.format("Specified trigger compact entryLocations: %s is invalid. "
+                                + "Bookie entry location RocksDB path: %s.", entryLocations, entryLocationDBPath);
+                        response.setCode(HttpServer.StatusCode.BAD_REQUEST);
+                    }
+                } else {
+                    // Not specified trigger compact entryLocations, trigger compact for all entry location.
+                    boolean isCompactRunning = ledgerStorage.isEntryLocationCompacting();
```

Comment:
We should trigger the compact which is not compacting. `path1`, `path2`, `path3`.
`path1`, and `path2` is compacting. we should ignore `path1` and `path2`. Trigger `path3` compact

Reply:
Changed the check logic, if user set `entryLocations` contain any rocksDB which already running compact, we will skip it and trigger other locations rocksDB compact.